### PR TITLE
A large test

### PR DESCRIPTION
**Pull request by mbrandonw**
Adding a test for a big html document so that we can exercise most of the library. we can add a lot more to this as we go and it will be a nice way of tracking how the API evolves.

### CODE REVIEW

File: Sources/HTML/Attributes.swift [hunk 1]
```diff
@@ -11,6 +11,7 @@ public let id = AttributeKey<String>(key: "id")
 public let height = AttributeKey<Int>(key: "height")
 public let max = AttributeKey<Int>(key: "max")
 public let maxlength = AttributeKey<Int>(key: "maxlength")
+public let method = AttributeKey<String>(key: "min")
```
Comment (stephencelis) [inline Sources/HTML/Attributes.swift]: Lil typo. Also do we wanna use something like `enum Method: Value { case get, post }` for things like this?

File: Sources/HTML/Elements.swift [hunk 2]
```diff
@@ -146,6 +186,18 @@ public func p(_ content: [Node]) -> Node {
...
+public func span(_ attribs: [Attribute], _ content: [Node]) -> Node {
+  return node("span", attribs, content)
+}
+
+public func span(_ content: [Node]) -> Node {
+  return span([], content)
+}
+
+public func title(_ string: String) -> Node {
+  return node("title", [], [.text(string)])
```
Comment (stephencelis) [inline Sources/HTML/Elements.swift]: Wondering if we should get rid of the `text` function now that we can use the enum case instead...

File: Tests/HTMLTests/FullDocumentTests.swift [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            title("The Site Title"),
+            style(
+              body |< (
+                background => rgb(240, 240, 240)
+              )
+            )
+          ]
+        ),
+        body(
+          [ HTML.`class` => "home",
```
Comment (stephencelis) [inline Tests/HTMLTests/FullDocumentTests.swift]: I know we're gonna try to figure out hot to not qualify this, but the nice thing about qualifying it here is we can lose the backticks! `HTML.class` compiles just fine :D

File: Tests/HTMLTests/HTMLTests.swift [hunk 4]
```diff
@@ -108,11 +101,4 @@ class HTMLTests: XCTestCase {
 
     XCTAssertEqual("<input disabled value=\"Hello, world!\"></input>", render(html))
   }
-
-  static var allTests = [
```
Comment (stephencelis) [inline Tests/HTMLTests/HTMLTests.swift]: :wave:

(Would be nice to get Linux tests running in the future, though!)

File: Sources/HTML/Attributes.swift (hunk 1 above)
Comment (mbrandonw) [inline Sources/HTML/Attributes.swift]: LOL wow. lil typo is an UNDERSTATEMENT.

yeah we can def do that. also we have `Method` repeated at least twice so maybe can consolidate some day

File: Sources/HTML/Elements.swift (hunk 2 above)
Comment (mbrandonw) [inline Sources/HTML/Elements.swift]: yes yes will do!

File: Tests/HTMLTests/FullDocumentTests.swift (hunk 3 above)
Comment (mbrandonw) [inline Tests/HTMLTests/FullDocumentTests.swift]: ohhohohohh!

File: Tests/HTMLTests/HTMLTests.swift (hunk 4 above)
Comment (mbrandonw) [inline Tests/HTMLTests/HTMLTests.swift]: lol man linux tests are such a pain

File: Sources/HTML/Attributes.swift (hunk 1 above)
Comment (stephencelis) [inline Sources/HTML/Attributes.swift]: I think it's ok to be separate. `HTML.Method` only supports `GET` and `POST`.

Comment (mbrandonw) [inline Sources/HTML/Attributes.swift]: oh right duh

**Comment by mbrandonw** [comment]
ok, I addressed all the feedback, will merge if tests pass!

**Pull request merged**